### ISSUE TITLE
prefer-const does not fix/report node.js require

### ISSUE THREAD

**Issue by deep9** [eslint/eslint#8184]
**Tell us about your environment**
* **ESLint Version:**
v3.16.1
* **Node Version:**
v6.9.2
* **npm Version:**
3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

```js
module.exports = {
    "env": {
        "es6": true,
        "node": true,
        "mocha": true
    },
    "extends": "eslint:recommended",
    "rules": {
        "indent": [
            "error",
            4,
            {
                "SwitchCase": 1
            }
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        // TODO: Ignoring quotes for now
        // "quotes": [
        //     "warn",
        //     "double",
        //     "avoid-escape"
        // ],
        "semi": [
            "error",
            "always"
        ],
        "no-unused-vars": [
            "error",
            {
                "argsIgnorePattern": "reject|db|helpers|next|_|req",
                "varsIgnorePattern": "logger"
            }
        ],
        "no-console": "off",
        "require-yield": "warn",
        "comma-dangle": [
            "error",
            {
                "arrays": "always-multiline",
                "objects": "always-multiline",
                "imports": "always-multiline",
                "exports": "always-multiline",
                "functions": "ignore",
            }
        ],
        "strict": [ "error", "global" ],
        "new-cap": [
            "error",
            {
                "capIsNewExceptions": [
                    "Router", // due to how express.Router is used
                ],
            },
        ],
        "keyword-spacing": [ "error" ],
        "prefer-const": [ "error" ],
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

```js
var utils = require('../utils'),
    apps = require('../models/apps'),
    stats = require('../models/stats'),
    router = require('express').Router();
const exceptions = require('../exceptions');
const appsModule = require('../apps');
var user = require('../models/user');
```

**What did you expect to happen?**
Expected that var XX = require() would be reported as candidate for const.

**What actually happened? Please include the actual, raw output from ESLint.**
Nothing.

**Comment by platinumazure** [comment]
According to the [`prefer-const` documentation](http://eslint.org/docs/rules/prefer-const), the rule only flags `let` declarations. You should use `no-var` if you want to catch `var` declarations.

The reason `prefer-const` doesn't flag `var` declarations is because of the hoisting and different scoping rules between `var` vs `let`/`const` declarations.

**Comment by deep9** [comment]
@platinumazure Thanks for the explanation!
